app/screens/manage_channel_members/manage_channel_members.tsx

```diff
@@ -231,25 +235,44 @@ export default function ManageChannelMembers({
 
     useNavButtonPressed(MANAGE_BUTTON, componentId, toggleManageEnabled, [toggleManageEnabled]);
 
-    useEffect(() => {
-        mounted.current = true;
-        const options: GetUsersOptions = {sort: 'admin', active: true, per_page: PER_PAGE_DEFAULT};
+    const handleReachedBottom = useCallback(() => {
+        if (hasMoreProfiles.current && !loading && !searchedTerm) {
+            pageRef.current += 1;
+            setLoading(true);
+            getFetchChannelMembers();
+        }
+    }, [loading, searchedTerm]);
+
+    const getFetchChannelMembers = useCallback(() => {
+        const options: GetUsersOptions = {sort: 'admin', active: true, per_page: PER_PAGE_DEFAULT, page: pageRef.current};
         fetchChannelMemberships(serverUrl, channelId, options, true).then(({users, members}) => {
             if (!mounted.current) {
                 return;
             }
-
-            if (users.length >= PER_PAGE_DEFAULT) {
-                hasMoreProfiles.current = true;
+            if (users.length < PER_PAGE_DEFAULT) {
+                hasMoreProfiles.current = false;
             }
             if (users.length) {
-                setProfiles(users);
-                setChannelMembers(members);
+                // eslint-disable-next-line max-nested-callbacks
+                setChannelMembers((prev) => [...prev, ...members]);
+                // eslint-disable-next-line max-nested-callbacks
+                setProfiles((prev) => [...prev, ...users]);
             }
+
             setLoading(false);
         });
+    }, []);
```

Comment:
Now that I look at this, this callback still depends on `serverUrl` and `channelId`. I don't expect them to change during the lifetime of the component, but still.

Either add them to the dependencies here, or make this callback receive them as arguments, and add them to the caller's dependencies.